### ISSUE TITLE
EnTK 0.72.0 breaks over pipeline.suspend()

### ISSUE THREAD

**Issue by SrinivasMushnoori** [radical-cybertools/radical.entk#375]
Looks like #296  has resurfaced.

[Here's a script that uses the suspend functionality](https://github.com/radical-cybertools/radical.entk/files/3637920/dummy_exchange.zip)

Running this gives me the following error:

```
Exception in thread dequeue-thread:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/scm177/VirtualEnvs/repex_devel/local/lib/python2.7/site-packages/radical/entk/appman/wfprocessor.py", line 435, in _dequeue
    self._update_dequeued_task(deq_task)
  File "/home/scm177/VirtualEnvs/repex_devel/local/lib/python2.7/site-packages/radical/entk/appman/wfprocessor.py", line 359, in _update_dequeued_task
    self._execute_post_exec(stage)
  File "/home/scm177/VirtualEnvs/repex_devel/local/lib/python2.7/site-packages/radical/entk/appman/wfprocessor.py", line 385, in _execute_post_exec
    stage.post_exec()
  File "dummy_exchange.py", line 330, in check_resume
    return self._check_res(self)
  File "dummy_exchange.py", line 226, in _check_resume
    _replica.resume()
  File "/home/scm177/VirtualEnvs/repex_devel/local/lib/python2.7/site-packages/radical/entk/pipeline/pipeline.py", line 305, in resume
    'Cannot resume Pipeline %s since it is not suspended' % self._uid)
EnTKError: Cannot resume Pipeline pipeline.0000 since it is not suspended
```

Which is the same issue as before: the `pipeline.suspend()` line is being reached, but isn't being executed. I am running the following stack:

```
(Env_RepEx)scm177@mcewan:~/RADICAL-LAB/repex_3/examples$ radical-stack

  python               : 2.7.14
  pythonpath           : 
  virtualenv           : /home/scm177/VirtualEnvs/Env_RepEx

  radical.analytics    : 0.72.0
  radical.entk         : 0.72.0
  radical.pilot        : 0.72.0
  radical.saga         : 0.72.0
  radical.utils        : 0.72.0
```
When I downgrade `radical.entk` to `0.7.17` I am able to run my workloads. 
[DEBUG level log file is here](https://github.com/radical-cybertools/radical.entk/files/3637937/debug_log.zip.zip)

This ticket has been somewhat overdue. Apologies. Thanks!

**Comment by andre-merzky** [comment]
Can we debug that after the transition to the `radical.repex` framework?  That makes tracking of replicas much easier (for me).

**Comment by andre-merzky** [comment]
It has literally resurfaced: the original fix got lost.  It's fixed again now in #376 .

**Comment by SrinivasMushnoori** [comment]
Okay: so should I upgrade or stay with 0.7.17?
I don't mind you fixing it after transition.

**Comment by andre-merzky** [comment]
The fix in #376 is not released yet.  In general the policy will be that the EnTK version configured in the `radical.repex` dependency should work for you, and there should not be a need to upgrade to a newer EnTK.  If an upgrade makes sense, I'll update that dependency, and you will get it automatically on the next new deployment.  So the info above is just to keep you posted, but there is no action needed from your side :-)
